nerfstudio/pipelines/base_pipeline.py

```diff
@@ -376,6 +376,9 @@ def get_average_eval_image_metrics(
             for camera, batch in self.datamanager.fixed_indices_eval_dataloader:
                 # time this the following line
                 inner_start = time()
+                if camera.metadata is not None:
+                    # fixes metrics with Splatfacto pose optimization and --eval-mode=all
+                    camera.metadata["cam_idx"] = idx
```

Comment:
This is something I am not sure if we have correct "cam_idx". Looking at the code, "cam_idx" key was set here: https://github.com/nerfstudio-project/nerfstudio/blob/main/nerfstudio/data/datamanagers/full_images_datamanager.py#L303

Are they the same? with the order of  `self.datamanager.fixed_indices_eval_dataloader`? I feel we really need to establish a global way to fetch indices of camera across different data managers. 

What about nerfacto case which uses `ParallelDataManager`?

Reply:
I did not actually test this with Nerfacto at all. This simply seemed to fix the metrics with `--eval mode all` in Splatfacto (if the indices were wrong, this would probably immediately destroy the PSNR metric).

> I feel we really need to establish a global way to fetch indices of camera across different data managers.

I fully agree :+1: This part of the PR is very fragile without such a mechanism

Reply:
On a second thought, this part of the PR is not critical and it could be safer to leave it out for now. The ability to export the optimized poses does not depend on this